stag-library/src/main/java/com/vimeo/stag/KnownTypeAdapters.java

```diff
@@ -336,27 +338,36 @@ public static void write(JsonWriter out, char value) throws IOException {
         }
     }
 
-
     /**
      * Helper class for {@link boolean}.
      */
     public static final class PrimitiveBooleanTypeAdapter {
 
+        private static final List<String> VALID_BOOLEAN_AS_STRING = Arrays.asList("true", "false", "1", "0");
+
         public static boolean read(JsonReader in, boolean defaultValue) throws IOException {
             JsonToken peek = in.peek();
             if (peek == JsonToken.NULL) {
                 in.nextNull();
                 return defaultValue;
-            } else if (peek == JsonToken.STRING) {
+            } else if (peek == JsonToken.STRING || peek == JsonToken.NUMBER) {
                 // support strings for compatibility with GSON 1.7
-                return Boolean.parseBoolean(in.nextString());
+                return Boolean.parseBoolean(booleanAsString(in.nextString()));
```

Comment:
As mentioned below, we can get rid of the `Boolean.parseBoolean` call if `booleanAsString` returned a boolean. In that case it should probably be renamed to `stringToBoolean`